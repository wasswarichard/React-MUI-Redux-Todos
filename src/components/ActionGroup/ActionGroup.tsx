import React from 'react';
import CardActions from "@mui/material/CardActions";
import Button from "@mui/material/Button";

interface actionGroupProps {
    disableDecline: boolean,
    disableApprove: boolean,
    declineTodo: (event : React.MouseEvent<HTMLButtonElement>) => void,
    approveTodo: (event : React.MouseEvent<HTMLButtonElement>) => void,
}

const ActionGroup : React.FC<actionGroupProps> = ({disableDecline, disableApprove, declineTodo, approveTodo}) => {
    return (
        <div>
            <hr/>
            <CardActions>
                {! disableDecline && <Button size="small" color="primary" onClick={declineTodo}>Decline</Button>}
                {! disableApprove && <Button  size="small" color="secondary" onClick={approveTodo}>Approve</Button>}
            </CardActions>
        </div>

    );
}
export default ActionGroup;