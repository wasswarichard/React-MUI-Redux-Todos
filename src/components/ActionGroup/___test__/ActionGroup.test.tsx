import React from "react";
import ActionGroup from "../ActionGroup";
import Enzyme, { shallow } from 'enzyme';
import {cleanup} from "@testing-library/react";
import Adapter from 'enzyme-adapter-react-16';
import TestRenderer from 'react-test-renderer';
Enzyme.configure({ adapter: new Adapter() })

describe('<ActionGroup>', () => {
    afterEach(cleanup)

    const statusFunction = () => {};
    it('renders BoardColumn Component', () => {
        shallow(<ActionGroup disableApprove={true} disableDecline={false} declineTodo={statusFunction} approveTodo={statusFunction}/>);
    });

    it('should match snapshot', function () {
        const tree = TestRenderer.create(<ActionGroup disableApprove={true} disableDecline={false} declineTodo={statusFunction} approveTodo={statusFunction}/>).toJSON();
        expect(tree).toMatchSnapshot();

    });
});