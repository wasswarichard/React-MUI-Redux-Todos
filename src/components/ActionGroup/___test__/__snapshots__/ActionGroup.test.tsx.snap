// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionGroup> should match snapshot 1`] = `
<div>
  <hr />
  <div
    className="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
  >
    <button
      className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root css-1knaqv7-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Decline
    </button>
  </div>
</div>
`;
